sdk/azidentity/azure_cli_credential.go

```diff
@@ -41,9 +43,9 @@ type AzureCLICredential struct {
 // options: configure the management of the requests sent to Azure Active Directory.
 func NewAzureCLICredential(options *AzureCLICredentialOptions) (*AzureCLICredential, error) {
 	if options == nil {
-		def := DefaultAzureCLICredentialOptions()
-		options = &def
+		options = &AzureCLICredentialOptions{}
 	}
+	options.init()
```

Comment:
You'll want to ensure that you `init()` a copy.  I suggest creating the `AzureCLICredential` type first then calling `init()` on its `options` field.

Reply:
The AzureCLICredential doesnt have the options struct on it. It has the TokenProvider directly on it, that's why I called the init before intializing it. I can create a copy of the options, initialize that and then assign it to the credential. Does that sounds like a good plan?

Reply:
Ah I see, I missed that.  Should it have the options in it?  E.g. if another option is added in the future.  If not your suggestion works fine.

Reply:
Hmm yes I remember.  Let's not change it now, we can in the future if the need arises.

Reply:
Done